bin/sozo/src/commands/options/account/controller.rs

```diff
@@ -0,0 +1,221 @@
+    let mut manifest_path = root_dir.to_path_buf();
+    manifest_path.extend(["manifests", profile, "base"]);
+    Ok(BaseManifest::load_from_path(&manifest_path)?)
+}
+
+fn collect_policies_from_base_manifest(
+    world_address: WorldAddressOrName,
+    user_address: FieldElement,
+    base_path: &Utf8Path,
+    manifest: BaseManifest,
+) -> Result<Vec<Policy>> {
+    let mut policies: Vec<Policy> = Vec::new();
+    let base_path: Utf8PathBuf = base_path.to_path_buf();
+
+    // compute the world address here if it's a name
+    let world_address = get_dojo_world_address(world_address, &manifest)?;
+
+    // get methods from all project contracts
+    for contract in manifest.contracts {
+        let contract_address = get_dojo_contract_address(world_address, &contract);
+        let abis = contract.inner.abi.unwrap().load_abi_string(&base_path)?;
+        let abis = serde_json::from_str::<Vec<AbiEntry>>(&abis)?;
+        policies_from_abis(&mut policies, &contract.name, contract_address, &abis);
+    }
+
+    // get method from world contract
+    let abis = manifest.world.inner.abi.unwrap().load_abi_string(&base_path)?;
+    let abis = serde_json::from_str::<Vec<AbiEntry>>(&abis)?;
+    policies_from_abis(&mut policies, &manifest.world.name, world_address, &abis);
+
+    // for sending declare tx
+    let method = "__declare_transaction__".to_string();
+    policies.push(Policy { target: user_address, method });
+    trace!("Adding declare transaction policy");
+
+    // for deploying using udc
+    let method = "deployContract".to_string();
+    policies.push(Policy { target: *UDC_ADDRESS, method });
+    trace!("Adding UDC deployment policy");
+
+    Ok(policies)
+}
+
+/// Recursively extract methods and convert them into policies from the all the
+/// ABIs in the project.
+fn policies_from_abis(
+    policies: &mut Vec<Policy>,
+    contract_name: &str,
+    contract_address: FieldElement,
+    entries: &[AbiEntry],
+) {
+    for entry in entries {
+        match entry {
+            AbiEntry::Function(f) => {
+                let method = f.name.to_string();
+                let policy = Policy { target: contract_address, method };
+                trace!(name = contract_name, target = format!("{:#x}", policy.target), method = %policy.method, "Adding policy");
+                policies.push(policy);
+            }
+
+            AbiEntry::Interface(i) => {
+                policies_from_abis(policies, contract_name, contract_address, &i.items)
+            }
+
+            _ => {}
+        }
+    }
+}
+
+fn get_dojo_contract_address(
+    world_address: FieldElement,
+    manifest: &Manifest<DojoContract>,
+) -> FieldElement {
+    if let Some(address) = manifest.inner.address {
+        address
+    } else {
+        let salt = generate_salt(&manifest.name);
+        get_contract_address(salt, manifest.inner.base_class_hash, &[], world_address)
+    }
```

Comment:
since we are reading from `BaseManifest` this field would't be set anytime, but to make sure that we don't depend on this behaviour (say we make the value FieldElement::ZERO by default for whatever reason) lets not check this and always calculate this.